ext/geometry/01_geometry.js

```diff
@@ -0,0 +1,2082 @@
+    webidl.assertBranded(this, DOMQuadPrototype);
+    return this[_p3];
+  }
+  get p4() {
+    webidl.assertBranded(this, DOMQuadPrototype);
+    return this[_p4];
+  }
+
+  getBounds() {
+    webidl.assertBranded(this, DOMQuadPrototype);
+    const { x: p1x, y: p1y } = this[_p1];
+    const { x: p2x, y: p2y } = this[_p2];
+    const { x: p3x, y: p3y } = this[_p3];
+    const { x: p4x, y: p4y } = this[_p4];
+
+    const left = MathMin(p1x, p2x, p3x, p4x);
+    const top = MathMin(p1y, p2y, p3y, p4y);
+    const right = MathMax(p1x, p2x, p3x, p4x);
+    const bottom = MathMax(p1y, p2y, p3y, p4y);
+
+    const bounds = webidl.createBranded(DOMRect);
+    bounds[_writable] = true;
+    bounds[_raw] = new Float64Array([
+      left,
+      top,
+      right - left,
+      bottom - top,
+    ]);
+    return bounds;
+  }
+
+  toJSON() {
+    webidl.assertBranded(this, DOMQuadPrototype);
+    return {
+      p1: this[_p1],
+      p2: this[_p2],
+      p3: this[_p3],
+      p4: this[_p4],
+    };
+  }
+
+  [SymbolFor("Deno.privateCustomInspect")](inspect, inspectOptions) {
+    return inspect(
+      createFilteredInspectProxy({
+        object: this,
+        evaluate: ObjectPrototypeIsPrototypeOf(DOMQuadPrototype, this),
+        keys: [
+          "p1",
+          "p2",
+          "p3",
+          "p4",
+        ],
+      }),
+      inspectOptions,
+    );
+  }
+}
+
+webidl.configureInterface(DOMQuad);
+const DOMQuadPrototype = DOMQuad.prototype;
+
+/**
+ * NOTE: column-major order
+ *
+ * For a 2D 3x2 matrix, the index of properties in
+ * | a c 0 e |    | 0 4 _ 12 |
+ * | b d 0 f |    | 1 5 _ 13 |
+ * | 0 0 1 0 | is | _ _ _  _ |
+ * | 0 0 0 1 |    | _ _ _  _ |
+ */
+const _a = 0;
+const _b = 1;
+const _c = 4;
+const _d = 5;
+const _e = 12;
+const _f = 13;
+
+/**
+ * NOTE: column-major order
+ *
+ * The index of properties in
+ * | m11 m21 m31 m41 |    | 0 4  8 12 |
+ * | m12 m22 m32 m42 |    | 1 5  9 13 |
+ * | m13 m23 m33 m43 | is | 2 6 10 14 |
+ * | m14 m24 m34 m44 |    | 3 7 11 15 |
+ */
+const _m11 = 0;
+const _m12 = 1;
+const _m13 = 2;
+const _m14 = 3;
+const _m21 = 4;
+const _m22 = 5;
+const _m23 = 6;
+const _m24 = 7;
+const _m31 = 8;
+const _m32 = 9;
+const _m33 = 10;
+const _m34 = 11;
+const _m41 = 12;
+const _m42 = 13;
+const _m43 = 14;
+const _m44 = 15;
+
+const _is2D = Symbol("[[is2D]]");
+
+class DOMMatrixReadOnly {
+  [_writable] = false;
+  [_raw];
+  [_is2D];
+
+  constructor(init = undefined) {
+    const prefix = `Failed to construct '${this.constructor.name}'`;
+    this[_brand] = _brand;
+    if (init === undefined) {
+      // deno-fmt-ignore
+      this[_raw] = new Float64Array([
+        1, 0, 0, 0,
+        0, 1, 0, 0,
+        0, 0, 1, 0,
+        0, 0, 0, 1,
+      ]);
+      this[_is2D] = true;
+    } else if (ObjectPrototypeIsPrototypeOf(DOMMatrixReadOnlyPrototype, init)) {
+      // NOTE: not in the spec, but requested by wpt
+      initMatrixFromMatrix(this, init);
+    } else if (webidl.type(init) === "Object") {
```

Comment:
This branch should only be taken if the object has a `[Symbol.iterator]` method according to the [algorithm to convert a JS value to an IDL union](https://webidl.spec.whatwg.org/#js-union). `new DOMMatrix({ toString: () => "" })` returns the identity matrix.